### PR TITLE
Update repository structure of source directory.

### PR DESCRIPTION
**Pull request by counter2015**
Improve the project structure documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,9 +56,19 @@ scala/
 +--lib/                      Pre-compiled libraries for the build
 +--src/                      All sources
    +---/library              Scala Standard Library
+   +---/library-aux          Auxiliary library for bootstrapping and documentation purposes
    +---/reflect              Scala Reflection
    +---/compiler             Scala Compiler
+   +---/interactive          Interface of interactive compiler to a client such as an IDE
    +---/intellij             IntelliJ project templates
+   +---/manual               Manual pages for runner scripts
+   +---/partest              Parser tests
```
Comment (dwijnand) [inline README.md]:
```suggestion
   +---/partest              Scala's internal parallel testing framework
```

File: README.md [hunk 2]
```diff
@@ -56,9 +56,19 @@ scala/
...
+   +---/partest-javaagent    Scala parser tests java agent
```
Comment (dwijnand) [inline README.md]:
```suggestion
   +---/partest-javaagent    Partest's helper java agent
```

File: README.md [hunk 3]
```diff
@@ -56,9 +56,19 @@ scala/
...
+   +---/repl                 Scala REPL core
+   +---/repl-frontend        Scala REPL frontend
+   +---/scaladoc             Scala documentation generators
+   +---/scalap               Source of `scalap`
```
Comment (dwijnand) [inline README.md]:
```suggestion
   +---/scalap               Scala's class file decompiler 
```

File: README.md [hunk 4]
```diff
@@ -56,9 +56,19 @@ scala/
...
+   +---/interactive          Interface of interactive compiler to a client such as an IDE
    +---/intellij             IntelliJ project templates
+   +---/manual               Manual pages for runner scripts
+   +---/partest              Parser tests
+   +---/partest-javaagent    Scala parser tests java agent
+   +---/repl                 Scala REPL core
+   +---/repl-frontend        Scala REPL frontend
+   +---/scaladoc             Scala documentation generators
+   +---/scalap               Source of `scalap` 
+   +---/testkit              Scala test kit
```
Comment (dwijnand) [inline README.md]:
```suggestion
   +---/testkit              Scala's unit-testing kit
```

File: README.md [hunk 5]
```diff
@@ -56,9 +56,19 @@ scala/
...
    +---/reflect              Scala Reflection
    +---/compiler             Scala Compiler
+   +---/interactive          Interface of interactive compiler to a client such as an IDE
    +---/intellij             IntelliJ project templates
+   +---/manual               Manual pages for runner scripts
+   +---/partest              Parser tests
+   +---/partest-javaagent    Scala parser tests java agent
+   +---/repl                 Scala REPL core
+   +---/repl-frontend        Scala REPL frontend
+   +---/scaladoc             Scala documentation generators
```
Comment (dwijnand) [inline README.md]:
```suggestion
   +---/scaladoc             Scala's documentation tool
```

**Review by dwijnand — CHANGES_REQUESTED** [review]
Great initiative, thank you, @counter2015.

Here's are a few fixes and some (more subjective) tweaks.

**Comment by counter2015** [comment]
Thanks for reviews @dwijnand ! File updated.

**Review by counter2015 — COMMENTED** [review]
review changes.

**Review by dwijnand — APPROVED** [review]

**Review by SethTisue — APPROVED** [review]
thank you!

**Pull request merged**